src/PackageManagement.UI/Common/RestartRequestBar.xaml.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Windows;
+using System.Windows.Controls;
+using Microsoft.VisualStudio.Shell.Interop;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using VsBrushes = Microsoft.VisualStudio.Shell.VsBrushes;
+
+namespace NuGet.PackageManagement.UI
+{
+    /// <summary>
+    /// Interaction logic for RestartRequestBar.xaml
+    /// </summary>
+    public partial class RestartRequestBar : UserControl, INuGetProjectContext
+    {
+
+        private readonly IDeleteOnRestartManager _deleteOnRestartManager;
+
+        private readonly IVsShell4 _vsRestarter;
+
+        public PackageExtractionContext PackageExtractionContext { get; set; }
+
+        public ISourceControlManagerProvider SourceControlManagerProvider { get; }
+
+        public ProjectManagement.ExecutionContext ExecutionContext { get; }
+
+        public RestartRequestBar(IDeleteOnRestartManager deleteOnRestartManager, IVsShell4 vsRestarter)
+        {
+            InitializeComponent();
+            _deleteOnRestartManager = deleteOnRestartManager;
+            _vsRestarter = vsRestarter;
+
+            // Set DynamicResource binding in code
+            // The reason we can't set it in XAML is that the VsBrushes class come from either
+            // Microsoft.VisualStudio.Shell.10 or Microsoft.VisualStudio.Shell.11 assembly,
+            // depending on whether NuGet runs inside VS10 or VS11.
+            StatusMessage.SetResourceReference(TextBlock.ForegroundProperty, VsBrushes.InfoTextKey);
+            RestartBar.SetResourceReference(Border.BackgroundProperty, VsBrushes.InfoBackgroundKey);
+            RestartBar.SetResourceReference(Border.BorderBrushProperty, VsBrushes.ActiveBorderKey);
+        }
+
+        private void UserControl_Loaded(object sender, RoutedEventArgs e)
+        {
+            CheckForUnsuccessfulUninstall();
```

Comment:
UserControl_Loaded is called every time the restartrequestbar comes into focus. This is not an issue, right?
Please add a comment here, stating that, we want it.